server/src/main/java/org/opensearch/index/search/QueryStringQueryParser.java

```diff
@@ -717,20 +716,25 @@ private Query getWildcardQuerySingle(String field, String termStr) throws ParseE
             // effectively, we check if a field exists or not
             return existsQuery(field);
         }
+        String indexedNameField = field;
         Analyzer oldAnalyzer = getAnalyzer();
         try {
             MappedFieldType currentFieldType = queryBuilder.context.fieldMapper(field);
-            if (currentFieldType == null) {
-                return newUnmappedFieldQuery(field);
-            }
-            if (forceAnalyzer != null && (analyzeWildcard || currentFieldType.getTextSearchInfo().isTokenized())) {
-                setAnalyzer(forceAnalyzer);
-                return super.getWildcardQuery(currentFieldType.name(), termStr);
+            if (currentFieldType != null) {
+                // return newUnmappedFieldQuery(field);
+                setAnalyzer(forceAnalyzer == null ? queryBuilder.context.getSearchAnalyzer(currentFieldType) : forceAnalyzer);
```

Comment:
Since we do this in multiple places maybe add a private `getSearchAnalyzer(field)` method?

if/else would probably look cleaner

Reply:
done